floors/floors.go

```diff
@@ -0,0 +1,207 @@
+	catchAll         string = "*"
+	skipRateMin      int    = 0
+	skipRateMax      int    = 100
+	modelWeightMax   int    = 100
+	modelWeightMin   int    = 1
+	enforceRateMin   int    = 0
+	enforceRateMax   int    = 100
+)
+
+// EnrichWithPriceFloors checks for floors enabled in account and request and selects floors data from dynamic fetched if present
+// else selects floors data from req.ext.prebid.floors and update request with selected floors details
+func EnrichWithPriceFloors(bidRequestWrapper *openrtb_ext.RequestWrapper, account config.Account, conversions currency.Conversions) []error {
+
+	if bidRequestWrapper == nil || bidRequestWrapper.BidRequest == nil {
+		return []error{errors.New("Empty bidrequest")}
+	}
+
+	if isPriceFloorsDisabled(account, bidRequestWrapper) {
+		return []error{errors.New("Floors feature is disabled at account or in the request")}
+	}
+
+	floors, err := resolveFloors(account, bidRequestWrapper, conversions)
+
+	updateReqErrs := updateBidRequestWithFloors(floors, bidRequestWrapper, conversions)
+	updateFloorsInRequest(bidRequestWrapper, floors)
+	return append(err, updateReqErrs...)
+}
+
+// updateBidRequestWithFloors will update imp.bidfloor and imp.bidfloorcur based on rules matching
+func updateBidRequestWithFloors(extFloorRules *openrtb_ext.PriceFloorRules, request *openrtb_ext.RequestWrapper, conversions currency.Conversions) []error {
+	var (
+		floorErrList []error
+		floorVal     float64
+	)
+
+	if extFloorRules == nil || extFloorRules.Data == nil || len(extFloorRules.Data.ModelGroups) == 0 {
+		return []error{}
+	}
+
+	modelGroup := extFloorRules.Data.ModelGroups[0]
+	if modelGroup.Schema.Delimiter == "" {
+		modelGroup.Schema.Delimiter = defaultDelimiter
+	}
+
+	extFloorRules.Skipped = new(bool)
+	if shouldSkipFloors(modelGroup.SkipRate, extFloorRules.Data.SkipRate, extFloorRules.SkipRate, rand.Intn) {
+		*extFloorRules.Skipped = true
+		return []error{}
+	}
+
+	floorErrList = validateFloorRulesAndLowerValidRuleKey(modelGroup.Schema, modelGroup.Schema.Delimiter, modelGroup.Values)
+	if len(modelGroup.Values) > 0 {
+		for _, imp := range request.GetImp() {
+			desiredRuleKey := createRuleKey(modelGroup.Schema, request, imp)
+			matchedRule, isRuleMatched := findRule(modelGroup.Values, modelGroup.Schema.Delimiter, desiredRuleKey)
+			floorVal = modelGroup.Default
+			if isRuleMatched {
+				floorVal = modelGroup.Values[matchedRule]
+			}
+
+			// No rule is matched or no default value provided or non-zero bidfloor not provided
+			if floorVal == 0.0 {
+				continue
+			}
+
+			floorMinVal, floorCur, err := getMinFloorValue(extFloorRules, imp, conversions)
+			if err == nil {
+				floorVal = roundToFourDecimals(floorVal)
+				bidFloor := floorVal
+				if floorMinVal > 0.0 && floorVal < floorMinVal {
+					bidFloor = floorMinVal
+				}
+
+				imp.BidFloor = bidFloor
+				imp.BidFloorCur = floorCur
+
+				if isRuleMatched {
+					err = updateImpExtWithFloorDetails(imp, matchedRule, floorVal, imp.BidFloor)
```

Comment:
It might be better to handle the error here, and then re-write line 100 to be an `else if` to capture a potential error that comes from `getMinFloorValue()`. What do you think?

Reply:
handled error here, 
Previous version was restructured to make below 2 lines common 
```
	if err != nil {
		floorErrList = append(floorErrList, err)
	}
```